crates/rooch-framework/sources/coin_entry.move

```diff
@@ -0,0 +1,281 @@
+/// CoinEntry is built to make a simple walkthrough of the Coins module.
+/// It contains scripts you will need to initialize, mint, burn, transfer coins.
+/// By utilizing this current module, a developer can create his own coin and care less about mint and burn capabilities,
+module rooch_framework::coin_entry {
+    use std::error;
+    use std::signer;
+    use std::string;
+    use moveos_std::account_storage;
+    use moveos_std::storage_context::StorageContext;
+    use rooch_framework::coin::{BurnCapability, MintCapability, FreezeCapability};
+
+    use rooch_framework::coin;
+    use rooch_framework::account;
+
+    //
+    // Errors
+    //
+
+    /// account has no capabilities (burn/mint).
+    const ENoCapabilities: u64 = 1;
+
+    //
+    // Data structures
+    //
+
+    /// Capabilities resource storing mint and burn capabilities.
+    /// The resource is stored on the account that initialized coin `CoinType`.
+    struct Capabilities<phantom CoinType> has key {
+        burn_cap: BurnCapability<CoinType>,
+        freeze_cap: FreezeCapability<CoinType>,
+        mint_cap: MintCapability<CoinType>,
+    }
+
+    //
+    // Public functions
+    //
+
+    /// Initialize new coin `CoinType` in Rooch Blockchain.
+    /// Mint and Burn Capabilities will be stored under `account` in `Capabilities` resource.
+    public entry fun initialize<CoinType>(
+        ctx: &mut StorageContext,
+        account: &signer,
+        name: vector<u8>,
+        symbol: vector<u8>,
+        decimals: u8,
+    ) {
+        let (burn_cap, freeze_cap, mint_cap) = coin::initialize<CoinType>(
```

Comment:
Can the user directly call the `coin::initialize` function? Or the user needs to call the  CoinType's initialize function. such as `gas_coin::initialize`

Reply:
> Can the user directly call the `coin::initialize` function? Or the user needs to call the CoinType's initialize function. such as `gas_coin::initialize`

of course user can directly call `coin::initialize` function,  coin_entry encapsulates `coin::initialize`, and stores the caps under the user account.

In normal case, the Coin created by the user does not need an independent contract, just call `coin_entry::initialize` directly, of course can also call `coin::initialize`.

Reply:
> In normal case, the Coin created by the user does not need an independent contract

But the user still needs to deploy a contract to define the CoinType.

If the developer wants to do some extra init for the coin, like gas_coin. Let the coin developer wrap the init and entry functions is easy to do custom extensions.

We can make this TBD.